packages/forms/backend/sirius-components-forms-graphql/src/main/java/org/eclipse/sirius/components/forms/graphql/datafetchers/subscription/SubscriptionDiagramFilterEventDataFetcher.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.forms.graphql.datafetchers.subscription;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.sirius.components.annotations.spring.graphql.SubscriptionDataFetcher;
+import org.eclipse.sirius.components.collaborative.forms.api.DiagramFilterConfiguration;
+import org.eclipse.sirius.components.collaborative.forms.api.IFormEventProcessor;
+import org.eclipse.sirius.components.collaborative.forms.dto.PropertiesEventInput;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.graphql.api.IDataFetcherWithFieldCoordinates;
+import org.eclipse.sirius.components.graphql.api.IEventProcessorSubscriptionProvider;
+import org.eclipse.sirius.components.graphql.api.IExceptionWrapper;
+import org.eclipse.sirius.components.graphql.api.LocalContextConstants;
+import org.reactivestreams.Publisher;
+
+import graphql.execution.DataFetcherResult;
+import graphql.schema.DataFetchingEnvironment;
+
+/**
+ * The data fetcher used to send the refreshed diagram filters to a subscription.
+ *
+ * @author gdaniel
+ */
+@SubscriptionDataFetcher(type = "Subscription", field = "diagramFilterEvent")
+public class SubscriptionDiagramFilterEventDataFetcher implements IDataFetcherWithFieldCoordinates<Publisher<DataFetcherResult<IPayload>>> {
+
+    private static final String INPUT_ARGUMENT = "input";
+
+    private final ObjectMapper objectMapper;
+
+    private final IExceptionWrapper exceptionWrapper;
+
+    private final IEventProcessorSubscriptionProvider eventProcessorSubscriptionProvider;
+
+    public SubscriptionDiagramFilterEventDataFetcher(ObjectMapper objectMapper, IExceptionWrapper exceptionWrapper, IEventProcessorSubscriptionProvider eventProcessorSubscriptionProvider) {
+        this.objectMapper = Objects.requireNonNull(objectMapper);
+        this.exceptionWrapper = Objects.requireNonNull(exceptionWrapper);
+        this.eventProcessorSubscriptionProvider = Objects.requireNonNull(eventProcessorSubscriptionProvider);
+    }
+
+    @Override
+    public Publisher<DataFetcherResult<IPayload>> get(DataFetchingEnvironment environment) throws Exception {
+        Object argument = environment.getArgument(INPUT_ARGUMENT);
+        var input = this.objectMapper.convertValue(argument, PropertiesEventInput.class);
```

Comment:
I used `PropertiesEventInput` because in the frontend I rely on `FormBasedView` to create the view and open the subscription. The type of input in hardcoded in `FormBasedView`, so I didn't have an easy way to change it and still rely on `FormBasedView`.
To support this we can either:
- Add a props in `FormBasedView` for the input we want to use to open the subscription (this will probably have some impacts in other parts of the code)
- Create a new "FormBasedView" tailored to the diagram filter that would create the appropriate input (in this case we will have quite a lot of duplicated code between the two views)

Reply:
We discussed it, we'll keep `PropertiesEventInput` for now